org.scala.tools.eclipse.search/src/org/scala/tools/eclipse/search/indexing/Occurrence.scala

```diff
@@ -39,25 +40,27 @@ case class Occurrence(
     offset: Int, // char offset from beginning of file
     occurrenceKind: OccurrenceKind,
     lineContent: String = "",
-    isInSuperPosition: Boolean = false) {
+    isInSuperPosition: Boolean = false,
+    callerOffset: Option[Int] = None) {
 
   override def equals(other: Any) = other match {
     // Don't compare lineCOntent
     case o: Occurrence =>
       word == o.word &&
-      file == o.file &&
-      offset == o.offset &&
-      occurrenceKind == o.occurrenceKind &&
-      isInSuperPosition == o.isInSuperPosition
+        file == o.file &&
+        offset == o.offset &&
+        occurrenceKind == o.occurrenceKind &&
+        isInSuperPosition == o.isInSuperPosition &&
+        callerOffset == o.callerOffset
     case _ => false
```

Comment:
Do we need the custom `equals` method at all? `lineContent` doesn't seem to be used here. You could use a second parameter list like `case class Occurrence(...)(lineContent: String = "")` to avoid that the compiler includes it in the generated `equals` method.